core/src/main/java/com/google/zxing/datamatrix/encoder/HighLevelEncoder.java

```diff
@@ -164,6 +167,13 @@ public static String encodeHighLevel(String msg,
     }
 
     int encodingMode = ASCII_ENCODATION; //Default mode
+
+    if (forceC40) {
```

Comment:
Maybe put this init at the top, or even make a C40Encoder overload that carries the 'force' info so it can be used internally. I'd rather hide, I think, this detail inside the encoder rather than make callers do the work if possible. Like can C40Encoder call encodeMaximal internally?

and is forcing C40 really kinda synonymous with 'maximal' encoding?

Reply:
>Like can C40Encoder call encodeMaximal internally

I don't see how but perhaps you have an idea. The problem is that we have to start in C40 encoding. If we act from inside C40Encoder.encode(), then we will get called when the lookahead test thinks that C40 encoding is appropriate. I was quite happy with the solution since it is very clear so see what happens if forceC40 is false (the important path). There is only one if that prunes the entire right at the start. 

>and is forcing C40 really kinda synonymous with 'maximal' encoding?
No, but the C40Encoder doesn't force c40 encoding. Instead it encodes maximally in the sense that it is greedy and encodes as much as it can when this function is called. In contrast, the normal encode() function just encodes on triplet and then consults the lookahead test to see if it should encode more.

Reply:
OK I think it's fine if this is the cleanest solution